staging/src/k8s.io/client-go/tools/remotecommand/remotecommand.go

```diff
@@ -122,6 +123,11 @@ func (e *streamExecutor) Stream(options StreamOptions) error {
 	}
 	defer conn.Close()
 
+	go func() {
```

Comment:
I think this PR change can't achieve the effect you want, but an extra goroutine leak

Reply:
> goroutine leak If CloseChan is not set

I found that the `Connection` interface provides a `CloseChan()`. Using `CloseChan()` can fix the problem of goroutine leakage.

```go
// Connection represents an upgraded HTTP connection.
type Connection interface {
	// CreateStream creates a new Stream with the supplied headers.
	CreateStream(headers http.Header) (Stream, error)
	// Close resets all streams and closes the connection.
	Close() error
	// CloseChan returns a channel that is closed when the underlying connection is closed.
	CloseChan() <-chan bool
	// SetIdleTimeout sets the amount of time the connection may remain idle before
	// it is automatically closed.
	SetIdleTimeout(timeout time.Duration)
}
```